app/routes.py

```diff
@@ -1,2 +1,328 @@
-from flask import Blueprint
+from app import db
+from app.models.customer import Customer
+from app.models.video import Video
+from app.models.rental import Rental
+from datetime import datetime
+from flask import Blueprint, request, jsonify
+from flask.helpers import make_response
+import os
+import requests
 
+# set up blueprints
+# wave 1 (customers & videos)
+videos_bp = Blueprint("videos", __name__, url_prefix="/videos")
+customers_bp = Blueprint("customers", __name__, url_prefix="/customers")
+# wave 2 (rentals)
+rentals_bp = Blueprint("rentals", __name__, url_prefix="/rentals")
+
+# helper function to check that customer/video ids are integers
+def is_int(value):
+    try:
+        return int(value)
+    except ValueError:
+        return False
+# **********************************************************
+# Wave 1: CRUD (Create, Read, Update & Delete) for customer
+# **********************************************************
+@customers_bp.route("", methods=["POST"], strict_slashes=False)
+def create_customer():
+    try:
+        customer_data = request.get_json()
+
+        new_customer = Customer(name=customer_data["name"],
+                                postal_code=customer_data["postal_code"],
+                                phone=customer_data["phone"],
+                                registered_at=datetime.now())
+        db.session.add(new_customer)
+        db.session.commit()
+
+        return jsonify({
+            "id": new_customer.customer_id
+        }), 201
+    except KeyError:
+        return make_response({"details": "Invalid data"}, 400)
+
+@customers_bp.route("", methods=["GET"], strict_slashes=False)
+def get_customer():
+    customer_name_from_url = request.args.get("name")
+    # get customer by name
+    if customer_name_from_url:
+        customers = Customer.query.filter.by(name=customer_name_from_url)
+    # get all customers
+    else:
+        customers = Customer.query.all()
+    
+    customers_response = []
+
+    for customer in customers:
+        customers_response.append(customer.to_json())
+    
+    return jsonify(customers_response), 200
+
+@customers_bp.route("/<customer_id>", methods=["GET"], strict_slashes=False)
+def get_one_customer(customer_id):
+    if not is_int(customer_id):
+        return {
+            "message": f"ID {customer_id} must be an integer",
+            "success": False
+        }, 400
+    
+    customer = Customer.query.get(customer_id)
+
+    if customer is None:
+        return make_response("Customer does not exist", 404)
+    else:
+        return customer.to_json(), 200
+
+@customers_bp.route("/<customer_id>", methods=["PUT"], strict_slashes=False)
+def update_customer(customer_id):
+    customer = Customer.query.get(customer_id)
+    customer_data = request.get_json()
+
+    if customer is None:
+        return make_response("Customer does not exist", 404)
```

Comment:
We can combine these lines into one using `get_or_404`. This can be used in place of `get` for any query made.

```suggestion
    customer = Customer.query.get_or_404(customer_id)
    customer_data = request.get_json()
```